tools/src/faircv/Main.hs

```diff
@@ -76,15 +76,22 @@ main = do
             }
         infoToNewSub :: AssignmentStudentInfo -> NewSubmission
         infoToNewSub = signedSubmissionToRequest . toSignedSubmission . infoToSubmission
+        waitForProofs :: [SubmissionStudentInfo] -> IO ()
+        waitForProofs [] = return ()
+        waitForProofs (info:infos) = do
+            newInfo <- sGetSubmission sClient $ siHash info
+            case giHasProof <$> siGrade newInfo of
+                Just True -> waitForProofs infos
+                _ -> threadDelay waitingTime >> waitForProofs (info:infos)
 
     -- Get all the available assignments for this student
     assLst <- sGetAssignments sClient Nothing Nothing Nothing False
     -- Make a new submission from the first one, then send it
-    mapM_ (sendSubmission sClient . infoToNewSub) $ take 1 assLst
+    subInfos <- mapM (sendSubmission sClient . infoToNewSub) $ take 1 assLst
 
-    -- Wait some time for the proofs to be available, then get them and make a faircv
+    -- Wait for the proofs to be available, then get them and make a faircv
     logInfo "Submissions sent, waiting for proofs"
-    threadDelay $ sec 20
+    waitForProofs subInfos
```

Comment:
:cool:

Reply:
IMO it may worth waiting for a little bit more (fractions of seconds) after proofs are found, since your ask educator for proofs, but then verify FairCV in witness. It may take a little while for the educator to send its block to the witness.